lib/rules/no-func-assign.js

```diff
@@ -13,6 +13,8 @@ const astUtils = require("../util/ast-utils");
 
 module.exports = {
     meta: {
+        type: "problem",
```

Comment:
I'd be fine with either `problem` or `suggestion` here, but I think it should be the same as `no-class-assign`. Currently, `no-class-assign` is `suggestion` and `no-func-assign` is `problem`.

Reply:
I agree, these should be consistent. They also seem like both are problems.